common/views/components/ZoomedImage/ZoomedImage.js

```diff
@@ -0,0 +1,212 @@
+// @flow
+import fetch from 'isomorphic-unfetch';
+import openseadragon from 'openseadragon';
+import { useState, useEffect, useRef } from 'react';
+import styled from 'styled-components';
+import { trackEvent } from '@weco/common/utils/ga';
+import Raven from 'raven-js';
+import Control from '@weco/common/views/components/Buttons/Control/Control';
+
+const ZoomedImageContainer = styled.div`
+  width: 100vw;
+  height: 100vh;
+  position: fixed;
+  top: 0;
+  left: 0;
+  z-index: 2;
+  background: ${props => props.theme.colors.black};
+`;
+
+const Image = styled.div`
+  height: 100%;
+`;
+
+const ErrorMessage = () => (
+  <div>
+    <p>The image viewer is not working.</p>
+  </div>
+);
+
+type Props = {|
+  id: string,
+  infoUrl: string,
+  setShowViewer: boolean => void,
+|};
+
+const ZoomedImage = ({ id, infoUrl, setShowViewer }: Props) => {
+  const [scriptError, setScriptError] = useState(false);
+  const [viewer, setViewer] = useState(null);
+  const zoomStep = 0.5;
+  const firstControl = useRef(null);
+  const lastControl = useRef(null);
+  const zoomedImage = useRef(null);
+  const handleScriptError = error => {
+    Raven.captureException(new Error(`OpenSeadragon error: ${error}`), {
+      tags: {
+        service: 'dlcs',
+      },
+    });
+    setScriptError(true);
+  };
+
+  function setupViewer(imageInfoSrc, viewerId, handleScriptError) {
```

Comment:
still unsure as to why `handleScriptError` is an parameter when it's always the same?